### PR TITLE
[plugin.video.twitch] 1.4.1

### PR DESCRIPTION
**Pull request by MrSprigster**
[plugin.video.twitch] 1.4.1

Added compatibility to reflect API Changes by Twitch

### CODE REVIEW

**Comment by learningit** [comment]
You need to squash these commits before it's reviewed. See https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

**Comment by MrSprigster** [comment]
My apologies, I thought I had already squashed this but I clearly hadn't.

Should be a single commit now. 

Thanks! :+1:

File: plugin.video.twitch/addon.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        <import addon="script.module.xbmcswift2" version="1.2.0"/>
+    </requires>
+    <extension point='xbmc.python.pluginsource' library='resources/lib/__run__.py'>
+        <provides>video</provides>
+    </extension>
+    <extension point='xbmc.addon.metadata'>
+        <platform>all</platform>
+        <language>en</language>
+        <source>https://github.com/StateOfTheArt89/Twitch.tv-on-XBMC</source>
+        <license>GNU GENERAL PUBLIC LICENSE Version 3, 29 June 2007</license>
```
Comment (learningit) [inline plugin.video.twitch/addon.xml]: would be good if you could add \<forum\> entry in your next update

File: plugin.video.twitch/resources/language/Czech/strings.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="yes"?>
+<strings>
```
Comment (learningit) [inline plugin.video.twitch/resources/language/Czech/strings.xml]: Please change to strings.po for your next update, strings.xml is now deprecated.

File: plugin.video.twitch/resources/lib/converter.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    def convertToXBMCPlaylist(InputPlaylist, title='', image=''):
+        # Create playlist in Kodi, return tuple (playlist, {listitem dict}) or ()
+        playlist = xbmc.PlayList(xbmc.PLAYLIST_VIDEO)
+        playlist.clear()
+        initialItem = None
+        for (url, details) in InputPlaylist:
+            if url:
+                if details != ():
+                    (title, image) = details
+                playbackItem = xbmcgui.ListItem(label=title, thumbnailImage=image, path=url)
```
Comment (learningit) [inline plugin.video.twitch/resources/lib/converter.py]: the thumbnailImage init field is now deprecated. Please change to listItem.setArt() in your next update.

File: plugin.video.twitch/resources/lib/converter.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+                playbackItem.setProperty('IsPlayable', 'true')
+                playlist.add(url, playbackItem)
```
Comment (learningit) [inline plugin.video.twitch/resources/lib/converter.py]: Please set the videolabel 'mediatype' to the correct value - newer skins need it to display the item correctly.

File: plugin.video.twitch/resources/lib/converter.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        year = video.get(Keys.CREATED_AT)[:4] if video.get(Keys.CREATED_AT) else ''
+        image = video.get(Keys.PREVIEW) if video.get(Keys.PREVIEW) else Images.VIDEOTHUMB
+        return {'label': video[Keys.TITLE],
+                'path': self.plugin.url_for(endpoint='playVideo', _id=video['_id'], quality='-2'),
+                'context_menu': [(self.plugin.get_string(30077), 'RunPlugin(%s)' %
+                                  self.plugin.url_for(endpoint='playVideo', _id=video['_id'], quality='-1'))],
+                'is_playable': True,
+                'icon': image,
+                'thumbnail': image,
+                'info': {'duration': str(duration), 'plot': plot, 'plotoutline': plot, 'tagline': plot,
```
Comment (learningit) [inline plugin.video.twitch/resources/lib/converter.py]: I guess here is where you would set 'mediatype'.

File: plugin.video.twitch/resources/lib/swiftwrap.py [hunk 6]
```diff
@@ -0,0 +1,678 @@
...
+            'label2': label2,
+            'iconImage': icon,
+            'thumbnailImage': thumbnail,
+            'path': path,
+        }
+        #kwargs = dict((key, val) for key, val in locals().items() if val is
+        #not None and key != 'self')
+        kwargs = dict((key, val) for key, val in kwargs.items()
+                      if val is not None)
+        self._listitem = xbmcgui.ListItem(**kwargs)
```
Comment (learningit) [inline plugin.video.twitch/resources/lib/swiftwrap.py]: This should also be fixed to use listItem.setArt() instead of the inits.

**Comment by learningit** [comment]
I'm merging this but please look over my comments, setting the 'mediatype' is probably the one that should be done sooner, rather than later as a couple of skins really need it to display things correctly. thanks.

**Pull request merged**